services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -41,61 +44,60 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
+import scala.collection.JavaConversions;
 
 
-/**
- * This sets up the producers used by Kafka Monitoring and some of the reported metrics.
- */
 public class ProduceService implements Service {
   private static final Logger LOG = LoggerFactory.getLogger(ProduceService.class);
   private static final String METRIC_GROUP_NAME = "produce-service";
 
   private final String _name;
   private final ProduceMetrics _sensors;
-  private final KMBaseProducer _producer;
-  private final ScheduledExecutorService _executor;
+  private volatile KMBaseProducer _producer;
```

Comment:
The producer and the executor service is recreated when new partitions are detected.  I often make members volatile when I expect them to be updated and read by different threads.  Since new threads are created when the producer is changed out probably this is not needed.  But it is also unlikely to affect performance.